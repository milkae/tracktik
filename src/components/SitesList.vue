<template>
  <ul class="sites-list">
    <SitesListItem v-for="site in sites" :key="site.id" :site="site" />
  </ul>
</template>

<script lang="ts">
import Vue, { PropOptions } from "vue";
import SitesListItem from "@/components/SitesListItem.vue";
import { Site } from "@/store/sites/types";

export default Vue.extend({
  name: "SitesList",

  props: {
    sites: {
      type: Array,
      required: true
    } as PropOptions<Site[]>
  },

  components: { SitesListItem }
});
</script>

<style scoped lang="scss">
.sites-list {
  padding: 0;
  margin: 0;
}
</style>
