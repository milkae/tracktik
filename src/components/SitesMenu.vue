<template>
  <div>
    <SitesMenuFilters :filters="options.filters" />
    <SitesMenuSearch :search="options.search" />
    <SitesMenuSort :sort="options.sort" />
  </div>
</template>

<script lang="ts">
import Vue from "vue";
import { mapState } from "vuex";
import SitesMenuFilters from "@/components/SitesMenuFilters";
import SitesMenuSearch from "@/components/SitesMenuSearch";
import SitesMenuSort from "@/components/SitesMenuSort";

export default Vue.extend({
  name: "SitesMenu",

  computed: mapState("sites", { options: state => state.sitesMenuOptions }),

  watch: {
    options: function(newOptions) {
      this.$emit("submitted", newOptions);
    }
  },

  components: { SitesMenuFilters, SitesMenuSearch, SitesMenuSort }
});
</script>
