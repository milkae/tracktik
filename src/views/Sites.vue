<template>
  <div>
    <SitesMenu @submitted="searchSites" />
    <SitesList v-if="sites.length" :sites="sites" />
  </div>
</template>

<script>
import { mapState } from "vuex";
import SitesList from "@/components/SitesList";
import SitesMenu from "@/components/SitesMenu";

export default {
  name: "Sites",
  components: {
    SitesList,
    SitesMenu
  },
  methods: {
    searchSites({ filters, query, sort }) {
      this.$store.dispatch("SEARCH_SITES", { filters, query, sort });
    }
  },
  computed: mapState(["sites"])
};
</script>
