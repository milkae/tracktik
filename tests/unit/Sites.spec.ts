import { shallowMount } from "@vue/test-utils";
import Sites from "@/views/Sites.vue";
import SitesList from "@/components/SitesList.vue";
import SitesMenu from "@/components/SitesMenu.vue";

describe("Sites", () => {
  const build = () => {
    const wrapper = shallowMount(Sites, {
      data: () => ({
        sites: []
      })
    });

    return {
      wrapper,
      sitesList: () => wrapper.find(SitesList),
      sitesMenu: () => wrapper.find(SitesMenu)
    };
  };

  it("renders the component", () => {
    const { wrapper } = build();

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders main child components", () => {
    const { sitesList, sitesMenu } = build();

    expect(sitesList().exists()).toBe(true);
    expect(sitesMenu().exists()).toBe(true);
  });

  it("passes a binded sites prop to sites list component", () => {
    const { wrapper, sitesList } = build();
    wrapper.setData({
      sites: [{ name: "Site name" }]
    });

    expect((sitesList().vm as any).sites).toBe((wrapper.vm as any).sites);
  });
});
