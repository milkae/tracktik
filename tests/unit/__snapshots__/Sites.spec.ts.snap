// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sites renders the component 1`] = `
<div>
  <sitesmenu-stub></sitesmenu-stub>
  <!---->
</div>
`;
